common/network-common/src/main/java/org/apache/spark/network/sasl/aes/AesCipher.java

```diff
@@ -0,0 +1,298 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.spark.network.util.ByteArrayReadableChannel;
+import org.apache.spark.network.util.ByteArrayWritableChannel;
+import org.apache.spark.network.util.TransportConf;
+
+/**
+ * AES cipher for encryption and decryption.
+ */
+public class AesCipher {
+  private static final Logger logger = LoggerFactory.getLogger(AesCipher.class);
+  public static final String ENCRYPTION_HANDLER_NAME = "AesEncryption";
+  public static final String DECRYPTION_HANDLER_NAME = "AesDecryption";
+  public static final int STREAM_BUFFER_SIZE = 1024 * 32;
+  public static final String TRANSFORM = "AES/CTR/NoPadding";
+
+  private final SecretKeySpec inKeySpec;
+  private final IvParameterSpec inIvSpec;
+  private final SecretKeySpec outKeySpec;
+  private final IvParameterSpec outIvSpec;
+  private Properties properties;
+
+  public AesCipher(
+      Properties properties,
+      byte[] inKey,
+      byte[] outKey,
+      byte[] inIv,
+      byte[] outIv) throws IOException {
+    this.properties = properties;
+    inKeySpec = new SecretKeySpec(inKey, "AES");
+    inIvSpec = new IvParameterSpec(inIv);
+    outKeySpec = new SecretKeySpec(outKey, "AES");
+    outIvSpec = new IvParameterSpec(outIv);
+  }
+
+  public AesCipher(AesConfigMessage configMessage) throws IOException  {
+    this(new Properties(), configMessage.inKey, configMessage.outKey,
+      configMessage.inIv, configMessage.outIv);
+  }
+
+  /**
+   * Create AES crypto output stream
+   * @param ch The underlying channel to write out.
+   * @return Return output crypto stream for encryption.
+   * @throws IOException
+   */
+  public CryptoOutputStream createOutputStream(WritableByteChannel ch) throws IOException {
+    return new CryptoOutputStream(TRANSFORM, properties, ch, outKeySpec, outIvSpec);
+  }
+
+  /**
+   * Create AES crypto input stream
+   * @param ch The underlying channel used to read data.
+   * @return Return input crypto stream for decryption.
+   * @throws IOException
+   */
+  public CryptoInputStream createInputStream(ReadableByteChannel ch) throws IOException {
+    return new CryptoInputStream(TRANSFORM, properties, ch, inKeySpec, inIvSpec);
+  }
+
+  /**
+   * Add handlers to channel
+   * @param ch the channel for adding handlers
+   * @throws IOException
+   */
+  public void addToChannel(Channel ch) throws IOException {
+    ch.pipeline()
+      .addFirst(ENCRYPTION_HANDLER_NAME, new AesEncryptHandler(this))
+      .addFirst(DECRYPTION_HANDLER_NAME, new AesDecryptHandler(this));
+  }
+
+  /**
+   * Create the configuration message
+   * @param conf is the local transport configuration.
+   * @return Config message for sending.
+   */
+  public static AesConfigMessage createConfigMessage(TransportConf conf) {
+    int keySize = conf.aesCipherKeySize();
+    Properties properties = new Properties();
+
+    try {
+      int paramLen = CryptoCipherFactory.getCryptoCipher(AesCipher.TRANSFORM, properties)
```

Comment:
You still haven't addressed this comment. Why are you using an empty config instead of the values in the `TransportConf` parameter? With your code, for example, there's no way for someone to disable the use of OpenSSL since you'll be using the default value in `CryptoCipherFactory`.

Basically you need something like `CryptoStreamUtils.toCryptoConf` for this code.